tests/tck-build-logic/src/main/groovy/org/graalvm/internal/tck/harness/TckExtension.java

```diff
@@ -351,4 +353,62 @@ List<String> getMetadataFileList(Path directory) throws IOException {
             return foundFiles;
         }
     }
+
+    String getLatestLibraryVersion(String libraryModule) {
+        try {
+            List<String> coordinates = List.of(libraryModule.split(":"));
+            String group = coordinates.get(0);
+            String artifact = coordinates.get(1);
+
+            File coordinatesMetadataIndex = new File("metadata/" + group + "/" + artifact +"/index.json");
+            ObjectMapper objectMapper = new ObjectMapper()
+                    .enable(SerializationFeature.INDENT_OUTPUT)
+                    .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+            List<MetadataVersionsIndexEntry> entries = objectMapper.readValue(coordinatesMetadataIndex, new TypeReference<>() {
+            });
+
+            List<String> allTested = new ArrayList<>();
+            for (MetadataVersionsIndexEntry entry : entries) {
+                allTested.addAll(entry.testedVersions());
+            }
+
+            if (allTested.isEmpty()) {
+                throw new IllegalStateException("Cannot find any tested version for: " + libraryModule);
+            }
+
+            allTested.sort(Comparator.comparing(VersionNumber::parse));
+            return allTested.get(allTested.size() - 1);
```

Comment:
this is just `getLast()`, right?
```suggestion
            return allTested.getLast();
```

Reply:
`getLast` method on the `ArrayList` was not present in `JDK17`. Therefore, if we ever run something on `JDK17` (and [we do that for our tests](https://github.com/oracle/graalvm-reachability-metadata/blob/master/tests/tck-build-logic/src/main/groovy/org.graalvm.internal.tck-harness.gradle#L97)) we can't use `getLast()`.